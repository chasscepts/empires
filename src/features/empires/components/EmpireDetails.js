import PropTypes from 'prop-types';
import styles from './EmpireDetails.module.css';
import Wrapper from './Wrapper';

const linkName = (link) => link.split('/').pop().replace('_', ' ');

export default function EmpireDetails({ empire }) {
  return (
    <Wrapper>
      <h2 className={styles.heading}>
        <span className={styles.name}>{empire.name}</span>
        <span className={styles.expansion}>{empire.expansion}</span>
      </h2>
      <div className={styles.lightBg}>
        <div className={styles.groupHeading}>Army Type</div>
        <div className={styles.item}>{empire.army_type}</div>
      </div>
      <div className={styles.darkBg}>
        <div className={styles.groupHeading}>Team Bonus</div>
        <div className={styles.item}>{empire.team_bonus}</div>
      </div>
      <div className={styles.lightBg}>
        <div className={styles.groupHeading}>Civilization Bonus</div>
        <div>
          {empire.civilization_bonus.map((bonus) => (
            <div className={styles.item} key={bonus}>{bonus}</div>
          ))}
        </div>
      </div>
      <div className={styles.darkBg}>
        <div className={styles.groupHeading}>Unique Unit</div>
        <div>
          {empire.unique_unit.map((unit) => (
            <div className={styles.itemLink} key={unit} title="Link disabled">{linkName(unit)}</div>
          ))}
        </div>
      </div>
      <div className={styles.lightBg}>
        <div className={styles.groupHeading}>Unique Technology</div>
        <div>
          {empire.unique_tech.map((tech) => (
            <div className={styles.itemLink} key={tech} title="Link disabled">{linkName(tech)}</div>
          ))}
        </div>
      </div>
    </Wrapper>
  );
}

EmpireDetails.propTypes = {
  empire: PropTypes.shape(
    {
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      expansion: PropTypes.string.isRequired,
      army_type: PropTypes.string.isRequired,
      unique_unit: PropTypes.arrayOf(PropTypes.string).isRequired,
      unique_tech: PropTypes.arrayOf(PropTypes.string).isRequired,
      team_bonus: PropTypes.string.isRequired,
      civilization_bonus: PropTypes.arrayOf(PropTypes.string).isRequired,
    },
  ).isRequired,
};
