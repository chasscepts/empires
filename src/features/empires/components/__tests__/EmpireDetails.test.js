import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import RouterWrapper from '../test_helpers/RouterWrapper';
import EmpireDetails from '../EmpireDetails';
import sample from '../test_helpers/sampleEmpires';

const empire = sample.civilizations[0];

function Details() {
  return <EmpireDetails empire={empire} />;
}

it('matches snapshot', () => {
  const tree = renderer
    .create(<RouterWrapper component={Details} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('AppHeader', () => {
  beforeEach(() => {
    render(<RouterWrapper component={Details} />);
  });

  it('displays the name', () => {
    expect(screen.getByText(empire.name)).toBeInTheDocument();
  });

  it('displays the expansion', () => {
    expect(screen.getByText(empire.expansion)).toBeInTheDocument();
  });
});
