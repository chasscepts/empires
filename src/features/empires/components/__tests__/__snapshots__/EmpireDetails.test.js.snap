// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container"
>
  <header
    className="header"
  >
    <h1>
      Age Of Empires II
    </h1>
    <nav>
      <a
        className="link"
        href="/"
        onClick={[Function]}
        role="link"
      >
        üè†
      </a>
    </nav>
  </header>
  <h2
    className="heading"
  >
    <span
      className="name"
    >
      Aztecs
    </span>
    <span
      className="expansion"
    >
      The Conquerors
    </span>
  </h2>
  <div
    className="lightBg"
  >
    <div
      className="groupHeading"
    >
      Army Type
    </div>
    <div
      className="item"
    >
      Infantry and Monk
    </div>
  </div>
  <div
    className="darkBg"
  >
    <div
      className="groupHeading"
    >
      Team Bonus
    </div>
    <div
      className="item"
    >
      Relics generate +33% gold
    </div>
  </div>
  <div
    className="lightBg"
  >
    <div
      className="groupHeading"
    >
      Civilization Bonus
    </div>
    <div>
      <div
        className="item"
      >
        Villagers carry +5
      </div>
      <div
        className="item"
      >
        Military units created 15% faster
      </div>
      <div
        className="item"
      >
        +5 Monk hit points for each Monastery technology
      </div>
      <div
        className="item"
      >
        Loom free
      </div>
    </div>
  </div>
  <div
    className="darkBg"
  >
    <div
      className="groupHeading"
    >
      Unique Unit
    </div>
    <div>
      <div
        className="itemLink"
        title="Link disabled"
      >
        jaguar warrior
      </div>
    </div>
  </div>
  <div
    className="lightBg"
  >
    <div
      className="groupHeading"
    >
      Unique Technology
    </div>
    <div>
      <div
        className="itemLink"
        title="Link disabled"
      >
        garland wars
      </div>
    </div>
  </div>
</div>
`;
